tests/transformer/aws_test.py

```diff
@@ -1,15 +1,33 @@
 """Tests for the AWS transformer."""
+import filecmp
 import os
 
+from cloudimagedirectory import transformer
 
-def test_transformer_aws():
-    """Run transformer end to end with AWS input data."""
-    assert 0 == os.system(
-        "poetry run cloudimagedirectory-transformer -f"
-        " ${PWD}/tests/transformer/testdata/input/raw/aws/af-south-1.json -op=${PWD} "
-        " -dp=${PWD}/tests/transformer/testdata -v output --filter.until=none"
+
+def test_aws_transformer_command(runner, tmp_path):
+    """Verify that we can transform AWS data."""
+    runner.invoke(
+        transformer.run,
+        [
+            "-f",
+            "tests/transformer/testdata/input/raw/aws/af-south-1.json",
+            "-op=.",
+            f"-dp={tmp_path}",
+            "-v",
+            "output",
+            "--filter.until=none",
+        ],
     )
-    assert 0 == os.system(
-        "diff ${PWD}/tests/transformer/testdata/expected/aws/af-south-1/rhel_6.10_hvm_x86_64_hourly2"
-        " ${PWD}/tests/transformer/testdata/output/aws/af-south-1/rhel_6.10_hvm_x86_64_hourly2"
+
+    # Ensure the directory was made.
+    assert os.path.isdir(f"{tmp_path}/output/aws/af-south-1")
+
+    # Get current directory
+    pwd = os.getcwd()
+
+    # Check image data by comparing the expected file and the output file byte by byte.
+    filecmp.cmp(
```

Comment:
Maybe a stupid question (might be handled by pytest magic I don't know about) but without asserting the returnvalue of filecmp, do these tests even fail on "false"?

Reply:
It's not a stupid question, I was asking myself the same at some point :) I tried it out and read through the documentation of filecmp :) 

In the [documentation ](https://docs.python.org/3/library/filecmp.html) they say that `filecmp.cmp` compares the two files and if they are equal they return true per default, if not they return false. For some reason then the test also failes, but you are right - I wonder why the return value is not ignored. Maybe I will make it more explicit by adding something like this: 
```suggestion
    assert filecmp.cmp(
```
What do you think?

Reply:
Nice thanks for the information! Yeah adding the assert Infront wouldn't hurt and the rest would be a bit explicit. I like that.